src/standard.rs

```diff
@@ -197,9 +194,14 @@ pub(crate) fn decrypt_document_core(
     Ok(document
         .into_iter()
         .map(|(label, ciphertext)| {
-            let dec_result =
-                ironcore_documents::aes::decrypt_detached_document(&dek, ciphertext.into());
-            dec_result.map(|c| (label, c.0))
+            // Further validation of the IronCore MAGIC will be done inside the function
+            if ciphertext.starts_with(&[3]) {
```

Comment:
I see that you're trying to do some up front checking here to preserve a reasonable error message. I couldn't come up with anything that was better that kept the good error message.

Instead of changing the structure, we should instead make the `V3` public (which I've done in my ironcore-documents PR). So we should use that instead.